Amplify/Categories/DataStore/Model/Collection/List+Model.swift

```diff
@@ -23,35 +23,12 @@ public class List<ModelType: Model>: Collection, Codable, ExpressibleByArrayLite
     public typealias ArrayLiteralElement = ModelType
 
     /// The array of `Element` that backs the custom collection implementation.
-    internal var elements: Elements
-
-    /// If the list represents an association between two models, the `associatedId` will
-    /// hold the information necessary to query the associated elements (e.g. comments of a post)
-    internal var associatedId: Model.Identifier?
-
-    /// The associatedField represents the field to which the owner of the `List` is linked to.
-    /// For example, if `Post.comments` is associated with `Comment.post` the `List<Comment>`
-    /// of `Post` will have a reference to the `post` field in `Comment`.
-    internal var associatedField: ModelField?
-
-    internal var limit: Int = 100
-
-    /// The current state of lazily loaded list
-    internal var state: LoadState = .pending
+    public var elements: Elements
```

Comment:
No reason not to use `private(set)`, in the latest iteration, i've moved the storage of `elements` into their respective subclasses. I think this is better so that all logic is in the subclasses and testing is isolated to the plugin, without having to know what the super class does to get the full picture of the implementation details.